### DISCUSSION TITLE
How can I Get 403 Forbidden for endpoint 'users'?

### DISCUSSION THREAD

**Discussion by shivam-methi** [community/community#114738] [Apps, API and Webhooks] [answered]
### Select Topic Area

Question

### Body

I have to verify a scenario when I need to verify the 403 forbidden status for endpoint 'user'. so I have created an OAuth token with No Scope. 
But still when I'm trying to hit the API "https://api.github.com/user" I get some response(public) 200 ok and when I use Incorrect credentials I get 401 Not Authorized. but still unable to get the 403 Forbidden (Expected). 
also tried providing a rate limit, but it also didn't work out.
Is there any way by which I can get it?

**End Point: https://api.github.com/user
Required: 403 Forbidden**

https://docs.github.com/en/rest/users/users?apiVersion=2022-11-28#get-the-authenticated-user
![image](https://github.com/community/community/assets/51210330/35921ce3-df06-4453-a4a1-93a22cb2e387)

**Comment by mapluisch** [comment, marked as answer]
Hey Shivam,

you get a HTTP/2 403 response once your API rate limit is exceeded (5000 requests). 

Add parameter `-i` to see how many requests remain until you hit your limit, i.e. `curl -i -L (and so on)`.

I've just tested it on my end - after hitting the `x-ratelimit-limit`, you receive a 403 along with the message 
```json
{"message":"API rate limit exceeded for user ID [removed]. If you reach out to GitHub Support for help, please include the request ID [removed].","documentation_url":"https://docs.github.com/rest/overview/rate-limits-for-the-rest-api"}
```

Please be aware that the rate limit is user ID specific and not token specific.

Hope that helps!

**Reply by shivam-methi** [reply]
Thank you @mapluisch
